### DISCUSSION TITLE
Options before tabs start

### DISCUSSION THREAD

**Discussion by parzival418** [mathuo/dockview#337] [Q&A]
Great library!  I am looking to switch over from my current implementation in my app. I have one quick question.  Is there any way to add a custom icon/integration to the left of all the tabs? Below is my current design and I have the menu icon before all the tabs start and would love to implement this with dockview.

<img width="1245" alt="Screenshot 2023-09-24 at 11 33 30 PM" src="https://github.com/mathuo/dockview/assets/2397603/d896c7ac-bcff-4d49-a4f6-4a7761c22baf">

Thanks!

**Comment by techbech** [comment]
I believe this is possible using header action components [header action components.](https://dockview.dev/docs/components/dockview#group-controls-panel) You can add a action button with `leftHeaderActionsComponent` on a dockview.

**Reply by parzival418** [reply]
I had thought this, but "These controls are rendered of the left and right side of the space to the right of the tabs in the header bar." So it only renders to the right of the tab set. I want to put it to the left of all the tabs. Thanks though!

**Reply by techbech** [reply]
Oh, yeah, you're right. Let's see what @mathuo says :slightly_smiling_face:

**Comment by mathuo** [comment]
Ah yes I see your example slightly differs from both the `leftHeaderActionComponents` and `rightHeaderActionComponents`.

Certainly something that can be added though as I can see how that could be very useful, which would be header Actions to the left of the tabs section. Not sure on a name for it yet.

**Reply by parzival418** [reply]
That would be awesome as it is a really convenient place for my main menu icon which expands out into a dropdown like in web VSCode with all my file actions etc.
